mailbox/backup/src/main/java/org/apache/james/mailbox/backup/Zipper.java

```diff
@@ -22,15 +22,28 @@
 import java.io.IOException;
 import java.util.List;
 
+import org.apache.commons.compress.archivers.ArchiveEntry;
 import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream;
+import org.apache.commons.io.IOUtils;
 import org.apache.james.mailbox.store.mail.model.MailboxMessage;
 
+import com.github.fge.lambdas.Throwing;
+
 public class Zipper implements Backup {
 
     @Override
     public void archive(List<MailboxMessage> messages, File destination) throws IOException {
         try (ZipArchiveOutputStream archiveOutputStream = new ZipArchiveOutputStream(destination)) {
+            messages.stream()
```

Comment:
doing a for-loop would remove most of the boilerplace

Reply:
Shouldn't we be able to consume a stream of messages? Let's avoid loading full mailboxes in memory, please!

Reply:
I started a discussion in our chat be change a bit our "no Stream as input" policy

Reply:
Discussion conclusion : it's ok to pass the Stream as input because it's a private+local method